frontends/cli/source/cli_frontend.d

```diff
@@ -109,24 +109,29 @@ DeveloperSession login(Device device, ADI adi, bool interactive) {
     // ...
 
     if (account) return null;
-    if (!interactive) {
+    string appleId = environment.get("APPLE_ID_USER");
+    string password = environment.get("APPLE_ID_PWD");
+    if(interactive && (appleId is null || password is null)) {
+        log.info("Please enter your account informations. They will only be sent to Apple servers.");
+        log.info("See it for yourself at https://github.com/Dadoum/Sideloader/");
+
+        write("Apple ID: ");
+        appleId = readln().chomp();
+        password = readPasswordLine("Password: ");
+    }
+
+    if (appleId is null || password is null) {
         log.error("You are not logged in. (use `sidestore login` to log-in, or add `-i` to make us ask you the account)");
         return null;
     }
 
-    log.info("Please enter your account informations. They will only be sent to Apple servers.");
-    log.info("See it for yourself at https://github.com/Dadoum/Sideloader/");
-
-    write("Apple ID: ");
-    string appleId = readln().chomp();
-    string password = readPasswordLine("Password: ");
-
     return DeveloperSession.login(
         device,
         adi,
         appleId,
         password,
         (sendCode, submitCode) {
+            if(!interactive) return;
```

Comment:
What happens here if it's not interactive? Is an error printed? If not you should add one I think.

Reply:
If you run non-interactive and the MFA code is requested this errors out - it's not the prettiest but it does the job - I'm unfamiliar with error handling in D so I went with what seems to work :D 

Happy to replace this with a more language specific paradigm if you could advise.

Reply:
That's not a problem if that prints an error that's understandable enough